### DISCUSSION TITLE
Filter within data structures

### DISCUSSION THREAD

**Discussion by myzinsky** [directus/directus#6953] [Archive]
I want to retrieve the revisions for a specific element in my table "questisons":

```
{
   "_and":[
      {
         "collection":{
            "_eq":"questions"
         }
      },
      {
         "data":{
            "id":{
               "_eq":"63"
            }
         }
      }
   ]
}
```

Thats how I'm doing it however directs answers with 
```
{"errors":[{"message":"\"directus_revisions.data\" is not a relational field","extensions":{"code":"INVALID_QUERY"}}]}
```

This is the structure:

<img width="501" alt="image" src="https://user-images.githubusercontent.com/1970554/126876899-28604475-701b-4b38-a508-8500aff351f0.png">

can me somebody explain what I'm doing wrong?

**Comment by myzinsky** [comment]
After reading some dock it seams that I need to use ```deep```!

```
{"data":{"_filter":{"id":{"_eq":63}}}}
```

but also this does not work, it just returns everything...

**Comment by myzinsky** [comment]
Okay solved:

```
{"_and":[{"collection":{"_eq":"questions"}}, {"item":{"_eq":"63"}}]
```

**Comment by rijkvanzanten** [comment]
You can't search relationally/nested in the `data` object, as it's a flat JSON object 👍🏻
